src/registry/date/validation/constraint-mode.tsx

```diff
@@ -0,0 +1,115 @@
+import {DateComponentSchema} from '@open-formulieren/types';
+import {useFormikContext} from 'formik';
+import {FormattedMessage, MessageDescriptor, defineMessage, useIntl} from 'react-intl';
+
+import {Select} from '@/components/formio';
+
+import {AllModes, AllPossibleConstraints, DateConstraintKey, NonEmptyModes} from './types';
+
+// Mappings of value-label to produce dropdown options. Note that you need to filter out
+// only the options relevant to the particular field.
+const ALL_DATE_CONSTRAINT_MODE_OPTIONS: Array<{
+  label: MessageDescriptor;
+  value: NonEmptyModes;
+}> = [
+  {
+    value: 'fixedValue',
+    label: defineMessage({
+      description: "Date constraint mode 'fixedValue' label",
+      defaultMessage: 'Fixed value',
+    }),
+  },
+  {
+    value: 'future',
+    label: defineMessage({
+      description: "Date constraint mode 'future' label",
+      defaultMessage: 'In the future',
+    }),
+  },
+  {
+    value: 'past',
+    label: defineMessage({
+      description: "Date constraint mode 'past' label",
+      defaultMessage: 'In the past',
+    }),
+  },
+  {
+    value: 'relativeToVariable',
+    label: defineMessage({
+      description: "Date constraint mode 'relativeToVariable' label",
+      defaultMessage: 'Relative to variable',
+    }),
+  },
+];
+
+const MODES_TO_EXCLUDE: Record<DateConstraintKey, NonEmptyModes[]> = {
+  minDate: ['past'],
+  maxDate: ['future'],
+};
+
+const DEFAULT_VALUES: {
+  [K in AllPossibleConstraints['mode']]: Omit<Extract<AllPossibleConstraints, {mode: K}>, 'mode'>;
+} = {
+  '': {},
+  fixedValue: {},
+  future: {includeToday: false},
+  past: {includeToday: false},
+  relativeToVariable: {
+    variable: 'now',
+    delta: {
+      years: null,
+      months: null,
+      days: null,
+    },
```

Comment:
I notice that since these fields are next to each other in the UI, they overlap with the tooltip:
![tooltip-and-date](https://github.com/open-formulieren/formio-builder/assets/19154114/07ebc967-7f20-4a82-bf78-9d6054b68cd4)